Sources/Sentry/SentrySessionReplayIntegration.m

```diff
@@ -44,11 +44,12 @@ - (BOOL)installWithOptions:(nonnull SentryOptions *)options
     }
 
     if (@available(iOS 16.0, tvOS 16.0, *)) {
+        SentryReplayOptions *replayOptions = options.experimental.sessionReplayOptions;
+
         BOOL shouldReplayFullSession =
-            [self shouldReplayFullSession:options.sessionReplayOptions.replaysSessionSampleRate];
+            [self shouldReplayFullSession:replayOptions.sessionSampleRate];
 
-        if (!shouldReplayFullSession
-            && options.sessionReplayOptions.replaysOnErrorSampleRate == 0) {
+        if (!shouldReplayFullSession && replayOptions.errorSampleRate == 0) {
```

Comment:
that's a great point, I've actually missed that we will run recorder for nothing if fullSession wasn't sampled, will add this to Android as well 👍